### DISCUSSION TITLE
Redhat 9

### DISCUSSION THREAD

**Discussion by cepete** [apache/hop#4601] [General]
Has anybody successfully run Hop on a Redhat server?  If so, what version and did you have to install any extra packages or configuration to get it to work.  The reason I ask, is because no matter what I do, or what version I try, I get the following message:  
./hop-gui.sh
Error starting the Hop GUI: org/apache/beam/sdk/metrics/MetricResults
java.lang.NoClassDefFoundError: org/apache/beam/sdk/metrics/MetricResults
	at java.base/java.lang.Class.getDeclaredFields0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredFields(Class.java:3297)
	at java.base/java.lang.Class.getDeclaredFields(Class.java:2371)
	at org.apache.hop.core.HopClientEnvironment.findDeclaredFields(HopClientEnvironment.java:138)
	at org.apache.hop.ui.hopgui.HopGuiEnvironment.initGuiPlugins(HopGuiEnvironment.java:91)
	at org.apache.hop.ui.hopgui.HopGuiEnvironment.init(HopGuiEnvironment.java:68)
	at org.apache.hop.ui.hopgui.HopGuiEnvironment.init(HopGuiEnvironment.java:53)
	at org.apache.hop.ui.hopgui.HopGui.main(HopGui.java:339)
Caused by: java.lang.ClassNotFoundException: Unable to load class 'org.apache.beam.sdk.metrics.MetricResults' in this classloader or in the parent
	at org.apache.hop.core.plugins.HopURLClassLoader.loadClass(HopURLClassLoader.java:97)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
	... 8 more
Caused by: java.lang.ClassNotFoundException: org.apache.beam.sdk.metrics.MetricResults
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:445)
	at org.apache.hop.core.plugins.HopURLClassLoader.loadClassFromThisLoader(HopURLClassLoader.java:67)
	at org.apache.hop.core.plugins.HopURLClassLoader.loadClass(HopURLClassLoader.java:92)

**Comment by bamaer** [comment]
what are your java version and hop version? if you run a desktop environment, which one? if you don't run a desktop environment, you can't run hop gui.

**Reply by cepete** [reply]
openjdk 17.0.13 2024-10-15 LTS
OpenJDK Runtime Environment (Red_Hat-17.0.13.0.11-1) (build 17.0.13+11-LTS)
OpenJDK 64-Bit Server VM (Red_Hat-17.0.13.0.11-1) (build 17.0.13+11-LTS, mixed mode, sharing)
Apache Hop 2.10.0
Gnome

**Reply by bamaer** [reply]
that all looks ok. not redhat, but I run hop on desktop Linux (gnome) all the time without any issues. especially the beam issue looks strange. feel free to create a ticket so we have a trace.

**Reply by cepete** [reply]
Will do.  It's strange, this same image and project work fine on Linux Mint, but will not work on any version of Redhat.

**Reply by gmitter-ef** [reply]
Don't know if it fits here, but using the first example of https://hop.apache.org/tech-manual/latest/docker-container.html#_how_to_run_the_container
gives us similar errors on a (GUI-less) RHEL 8.1 machine with apache/hop:2.10.0 image.  The 2.9.0 image works however.

General - ERROR: Unexpected error registering jar plugin file: file:/opt/hop/plugins/misc/projects/hop-misc-projects-2.10.0.jar
2024/11/22 07:05:22 - General - ERROR: java.lang.ClassNotFoundException: Unable to load class 'org.apache.hop.projects.project.ManageProjectsOptionPlugin' in this classloader or in the parent
2024/11/22 07:05:22 - General -         at org.apache.hop.core.plugins.HopURLClassLoader.loadClass(HopURLClassLoader.java:97)
2024/11/22 07:05:22 - General -         at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
2024/11/22 07:05:22 - General -         at org.apache.hop.core.plugins.BasePluginType.registerPluginJars(BasePluginType.java:586)
2024/11/22 07:05:22 - General -         at org.apache.hop.core.plugins.BasePluginType.searchPlugins(BasePluginType.java:119)
2024/11/22 07:05:22 - General -         at org.apache.hop.core.plugins.PluginRegistry.registerType(PluginRegistry.java:536)
2024/11/22 07:05:22 - General -         at org.apache.hop.core.plugins.PluginRegistry.init(PluginRegistry.java:519)
2024/11/22 07:05:22 - General -         at org.apache.hop.core.HopEnvironment.init(HopEnvironment.java:119)
2024/11/22 07:05:22 - General -         at org.apache.hop.core.HopEnvironment.init(HopEnvironment.java:76)
2024/11/22 07:05:22 - General -         at org.apache.hop.config.HopConfig.main(HopConfig.java:166)
2024/11/22 07:05:22 - General - Caused by: java.lang.ClassNotFoundException: org.apache.hop.projects.project.ManageProjectsOptionPlugin
2024/11/22 07:05:22 - General -         at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:445)
2024/11/22 07:05:22 - General -         at org.apache.hop.core.plugins.HopURLClassLoader.loadClassFromThisLoader(HopURLClassLoader.java:67)
2024/11/22 07:05:22 - General -         at org.apache.hop.core.plugins.HopURLClassLoader.loadClass(HopURLClassLoader.java:92)
2024/11/22 07:05:22 - General -         ... 8 more

**Reply by bamaer** [reply]
so what you're saying is that the same host system behaves differently when running 2.9.0 vs 2.10.0 containers, @gmitter-ef? 
strange, other than Java 17 vs Java 11 and the hop version upgrade, I don't we made any significant changes to the image. definitely ticket material if that's the case

**Reply by gmitter-ef** [reply]
On first sight this is what I observed yes, but I will try to analyse more deeply and create a ticket if the issue is still present then.

**Comment by gertwieland** [comment]
I know it's a very late answer, but since the topic is still open, I wanted to add a comment:
We are running Hop on Rocky 9, but strictly as (short-lived) Docker containers. I'd say we have an enterprise-level setup, and it works perfectly fine (including performance).
